optional/Optional.py

```diff
@@ -0,0 +1,106 @@
+from abc import abstractmethod
+from .compatible_abc import CompatibleABC
+
+
+def of(thing):
```

Comment:
this feels a bit odd to me.  I get that using the capital "O" Optional for filename it reads like you are doing:
```
Optional.of("Foo")
```
but that file naming is unconventional.  I still wonder about putting the static constructors into the Optional base class?